tests_e2e/orchestrator/lib/agent_test_loader.py

```diff
@@ -109,8 +109,9 @@ def _validate(self):
             if suite.location != '':
                 for suite_image in suite.images:
                     for image in self.images[suite_image]:
-                        if len(image.locations) > 0:
-                            if suite.location not in image.locations:
+                        # If the image has a location restriction, validate that it is available on the location the suite must run on
```

Comment:
Now image.locations can be specified on a per-cloud basis (as a dictionary with the cloud name as  key); see comments in pipeline.yml (or the wiki) for the syntax